server/sonar-server/src/main/java/org/sonar/server/favorite/ws/SearchAction.java

```diff
@@ -86,83 +89,78 @@ private static SearchRequest toWsRequest(Request request) {
   private SearchResults toSearchResults(SearchRequest request) {
     userSession.checkLoggedIn();
     try (DbSession dbSession = dbClient.openSession(false)) {
-      SearchResults.Builder builder = SearchResults.newBuilder(request);
-      addAuthorizedProjectUuids(dbSession, builder);
-      addFavorites(builder);
-      return builder.build();
+      List<ComponentDto> authorizedFavorites = favoriteFinder.list().stream()
```

Comment:
As `favoriteFinder#list()` is used only by this call, I wonder if it should not manage permission filtering.